### PR TITLE
[PROCS-3306] Implement basic e2e test for Process Agent on Windows

### PR DESCRIPTION
**Pull request by daniel-taf**
### What does this PR do?
Adds basic end-to-end tests for process checks in a Windows environment.

### Additional Notes

Finding command-line utilities to start processes that could generate load was surprisingly difficult. As of now, these tests start with running an antivirus scan as it will last the entire time and have some IO stats. One possible drawback of this is that the user for this process is SYSTEM and not Administrator. An alternative would be to start compressing a very large file using `tar.exe` as it's already included in Windows.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/tests/process/windows_test.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
```
Comment (robertjli) [inline test/new-e2e/tests/process/windows_test.go:18]: Nitpick (optional): It's a bit unintuitive, but the standard we've been following is that the `test-infra-definitions` repo would still be considered external (to this repo), so they should be grouped in the 2nd group

**Review by robertjli — APPROVED** [review]
Awesome work! 🎉 

After you fix the linter errors, one last thing I'd recommend is to go into your CI pipeline and run the `new-e2e-process-dev` job in the `e2e` section. This runs the e2e tests in the `process` package against the image from the pipeline, so I think you need to wait for the build steps to finish before doing so. If those pass, then I think we're good to go 👍

**Comment by daniel-taf** [comment]
> Awesome work! 🎉
> 
> After you fix the linter errors, one last thing I'd recommend is to go into your CI pipeline and run the `new-e2e-process-dev` job in the `e2e` section. This runs the e2e tests in the `process` package against the image from the pipeline, so I think you need to wait for the build steps to finish before doing so. If those pass, then I think we're good to go 👍

Job succeeded
https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/397889481

**Pull request merged**